sample/src/commonMain/kotlin/SineWave.kt

```diff
@@ -1,58 +1,34 @@
 package com.juul.krayon.sample
 
-import com.juul.krayon.color.darkSlateBlue
-import com.juul.krayon.color.steelBlue
-import com.juul.krayon.color.white
-import com.juul.krayon.element.CircleElement
-import com.juul.krayon.element.RootElement
-import com.juul.krayon.kanvas.Kanvas
-import com.juul.krayon.kanvas.Paint
-import com.juul.krayon.scale.domain
-import com.juul.krayon.scale.range
-import com.juul.krayon.scale.scale
-import com.juul.krayon.selection.append
-import com.juul.krayon.selection.asSelection
-import com.juul.krayon.selection.data
-import com.juul.krayon.selection.each
-import com.juul.krayon.selection.selectAll
-import com.juul.krayon.shape.line
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
 import kotlin.math.PI
 import kotlin.math.sin
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.ExperimentalTime
+import kotlin.time.TimeSource
 
-/** Sample loosely follows https://bl.ocks.org/mbostock/0533f44f2cfabecc5e3a, but is not yet feature complete. */
-internal fun <PAINT, PATH> Kanvas<PAINT, PATH>.renderSineWave(samples: Int = 50) {
-    // Generate a sine wave, with missing data every fifth point
-    val data = (0 until samples).map { i ->
+/** A flow that emits data for a continuously rolling sine wave. By default, completes a full cycle in 10 seconds. */
+@OptIn(ExperimentalTime::class)
+internal fun movingSineWave(
+    period: Duration = 10.seconds,
+    samples: Int = 50,
+): Flow<List<Point?>> = flow {
+    var offset = 0f
+    var time = TimeSource.Monotonic.markNow()
+    while (true) {
+        emit(sineWave(offset, samples))
+        offset += ((time.elapsedNow() / period) * 2 * PI).toFloat()
+        time = TimeSource.Monotonic.markNow()
+        delay(1) // make sure we don't absolutely run away with CPU usage
```

Comment:
Perhaps `yield()` would have similar effect?
I'm really not certain, I think `delay` is probably fine, just pointing out another possible mechanism incase you hadn't thought of it.

Reply:
Delay was chosen intentionally here. `yield` will make sure that other coroutines get a chance to run, but `delay(1)` gives the CPU a chance to idle.